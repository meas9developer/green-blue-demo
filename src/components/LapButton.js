import React from 'react'
import { StyleSheet, Text, TouchableHighlight } from 'react-native'
import PropTypes from 'prop-types'

const styles = StyleSheet.create({
  button: {
    borderWidth: 2,
    height: 100,
    width: 100,
    borderRadius: 50,
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const LapButton = ({onLapButtonPress}) => {
  return (
    <TouchableHighlight
      onPress={onLapButtonPress}
      style={styles.button}
    >
      <Text>Lap button</Text>
    </TouchableHighlight>
  )
}

LapButton.propTypes = {
  onLapButtonPress: PropTypes.func
}
LapButton.defaultProps = {}

export default LapButton

