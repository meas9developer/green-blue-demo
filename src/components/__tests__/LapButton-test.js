import React from 'react'
import LapButton from './../LapButton'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const rendered = renderer.create(<LapButton />).toJSON()
  expect(rendered).toBeTruthy()
  expect(rendered).toMatchSnapshot()
});
