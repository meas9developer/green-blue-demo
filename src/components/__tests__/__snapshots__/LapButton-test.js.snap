// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "borderWidth": 2,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Lap button
  </Text>
</View>
`;
